diff --git a/configure.ac b/configure.ac
index f68602d..26abeeb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -36,7 +36,7 @@ AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR(m4)
 AC_CONFIG_AUX_DIR([build-aux])
 
-AM_INIT_AUTOMAKE([foreign 1.11 -Wall silent-rules tar-pax no-dist-gzip dist-xz])
+AM_INIT_AUTOMAKE([foreign 1.11 silent-rules tar-pax no-dist-gzip dist-xz])
 AM_SILENT_RULES([yes])
 
 AC_SUBST(LIBCANBERRA_VERSION_INFO, [2:5:2])
@@ -54,7 +54,7 @@ fi
 
 AC_PROG_MKDIR_P
 AC_PROG_LN_S
-AM_PROG_AR
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 
 # CC
 
@@ -702,7 +702,9 @@ AM_CONDITIONAL([BUILTIN_NULL], [test "x$BUILTIN_NULL" = x1])
 AC_SUBST([CA_MAJOR],[ca_major])
 AC_SUBST([CA_MINOR],[ca_minor])
 
-GTK_DOC_CHECK(1.9)
+m4_ifdef([GTK_DOC_CHECK], [
+  GTK_DOC_CHECK(1.9)
+])
 
 ###################################
 #            Output               #
