diff --git a/src/pulse.c b/src/pulse.c
index e28a517..abb9882 100644
--- a/src/pulse.c
+++ b/src/pulse.c
@@ -927,6 +927,8 @@ int driver_play(ca_context *c, uint32_t id, ca_proplist *proplist, ca_finish_cal
 
                 for (;;) {
                         ca_bool_t canceled;
+                        ca_prop *it;
+                        ca_bool_t ignore_return_value = 0;
 
                         pa_threaded_mainloop_lock(p->mainloop);
 
@@ -941,6 +943,23 @@ int driver_play(ca_context *c, uint32_t id, ca_proplist *proplist, ca_finish_cal
                                 goto finish_locked;
                         }
 
+                        /* check if property exists for skipping return value */
+                        ca_mutex_lock(proplist->mutex);
+                        for (it = proplist->first_item; it; it = it->next_item) {
+                                if (strncmp(it->key, "canberra.ignore_return_value", it->nbytes) == 0) {
+                                        if (strcmp(CA_PROP_DATA(it), "1") == 0) {
+                                                ignore_return_value = 1;
+                                        }
+                                        break;
+                                }
+                        }
+                        ca_mutex_unlock(proplist->mutex);
+                        if (ignore_return_value) {
+                                pa_operation_unref(o);
+                                ret = CA_SUCCESS;
+                                goto finish_locked;
+                        }
+
                         for (;;) {
                                 pa_operation_state_t state = pa_operation_get_state(o);
 
